electron_app/src/DBManager.js

```diff
@@ -199,6 +187,38 @@ const updateContactByEmail = ({ email, name }, trx) => {
     .update({ name });
 };
 
+const updateContactScore = (emailId, trx) => {
+  const subquery = trx
+    .table(Table.EMAIL_CONTACT)
+    .select('contactId')
+    .where('emailId', emailId)
+    .andWhere('type', '<>', EMAIL_CONTACT_TYPE_FROM);
+  return trx
+    .table(Table.CONTACT)
+    .whereIn('id', subquery)
+    .increment('score', 1);
+};
+
+const updateContactSpamScore = ({ emailIds, notEmailAddress, value }) => {
+  return db
+    .table(Table.EMAIL_CONTACT)
+    .select(
+      db.raw(
+        `GROUP_CONCAT(DISTINCT(${Table.EMAIL_CONTACT}.contactId)) as contactIds`
+      )
+    )
+    .whereIn('emailId', emailIds)
+    .andWhere('type', EMAIL_CONTACT_TYPE_FROM)
+    .then(result => {
+      const contactIds = result[0].contactIds.split(',');
+      return db
+        .table(Table.CONTACT)
+        .whereNot('email', notEmailAddress)
+        .whereIn('id', contactIds)
+        .update({ spamScore: db.raw(`spamScore + ${value}`) });
```

Comment:
Could you use 
`.increment('spamScore', value)` 
... instead of ...
`.update({ spamScore: db.raw(`spamScore + ${value}`) })`
?
See:  https://knexjs.org/#Builder-increment